### ISSUE TITLE
Proposal: Support more border/background options on Panels

### ISSUE THREAD

**Issue by micahl** [Microsoft/microsoft-ui-xaml#518]
# Proposal: Support more border/background options on Panels

## Summary

Enable Panel (or a derived Panel) to support more options for borders and background.  Specifically, support:

- Per-edge border colors
- Per-corner radius
- Per-edge border thickness
- Support for showing a solid, dotted, or dashed border stroke
- The equivalent of [backfaceVisibility ](https://developer.mozilla.org/en-US/docs/Web/CSS/backface-visibility) in CSS (minimally 'visible' and 'hidden')

## Rationale

React Native's <View/> type supports a number of options for borders and background that in the Windows version do not map cleanly to any one element in XAML.  As a result, it must manage a combination of various elements to achieve the desired result.  Given the prevalence of <View/> in a RN-based app it's worth investigating the potential for reducing the # of native elements with the expectation of achieving better performance.

Related to https://github.com/Microsoft/react-native-windows/issues/2245.

## Functional Requirements

It could be that Panel exposes a number of protected methods to toggle these options and then any derived Panel-type can choose what options to expose.

**Comment by mdtauk** [comment]
Good rule of thumb for future additions to XAML...

If its possible in CSS, it should be possible in XAML.

**Comment by michael-hawker** [comment]
For `CornerRadius` is this about moving it to the base Panel?  As it already supports per-corner values, eh?

![image](https://user-images.githubusercontent.com/24302614/55754988-bedb7a80-5a02-11e9-9053-8ea3e6bf517f.png)

**Comment by micahl** [comment]
Yes, this idea would be about enabling the base Panel type to have built-in support for rendering something like borders such that a derived Panel could easily have that behavior.  This proposal is to track exploring whether that idea may or may not be useful for the React Native on Windows project.

CSS has been around a while and does a lot of things.  I'm not taking the default position that they're all good things to do.  :)

**Comment by dansmor7** [comment]
Any news on this?

Considering how the thickness and radius can be customized, I think color could work like this:
```XAML
BorderBrush="Red,Green,Yellow,Blue"
```
![image](https://user-images.githubusercontent.com/101892345/164951025-9cca5ec6-a665-4113-ae05-40d28730a38c.png)

I think it would also be useful if BackgroundSizing could be customized like that.

**Comment by michael-hawker** [comment]
@dansmor7's quick syntax is nice, though it's good to remember border brushes can be much more complex (static resources, bindings, image brushes, gradients, etc...). So, there'd be more work to do to enable those scenarios where an array of different brush types needs to be consumed and rendered.
